.github/autobuild/linux_deb.sh

```diff
@@ -61,18 +64,22 @@ setup() {
 }
 
 setup_cross_compilation_apt_sources() {
-    if [[ "${TARGET_ARCH}" == amd64 ]]; then
+    if [[ "${TARGET_ARCH}" == "${APT_ARCH}" ]]; then
         return
     fi
+    echo "Building on ${APT_ARCH} for ${TARGET_ARCH}. Thus modifying apt..."
     sudo dpkg --add-architecture "${TARGET_ARCH}"
-    # Duplicate the original Ubuntu sources and modify them to refer to the TARGET_ARCH:
-    sed -rne "s|^deb.*/ ([^ -]+(-updates)?) main.*|deb [arch=${TARGET_ARCH}] http://ports.ubuntu.com/ubuntu-ports \1 main universe multiverse restricted|p" /etc/apt/sources.list | sudo dd of=/etc/apt/sources.list.d/"${TARGET_ARCH}".list
-    # Now take the original Ubuntu sources and limit those to the build host (i.e. non-TARGET_ARCH) architectures:
-    sudo sed -re 's/^deb /deb [arch=amd64,i386] /' -i /etc/apt/sources.list
+
+    if [[ "${APT_ARCH}" == "amd64"  ]]; then
+        # Duplicate the original Ubuntu sources and modify them to refer to the TARGET_ARCH:
```

Comment:
Is there a reference to where this magic comes from?  I _think_ I can see what it's doing but it would be good if there were a "standard procedure" to follow.

Reply:
https://github.com/jamulussoftware/jamulus/blob/cc78e4c3cb45394f7513f67d1e7ce4f368beda27/.github/autobuild/linux_deb.sh#L54

Gives the architecture. For the repo replacing, you'd need to ask @hoffie 
The idea is to have more than one architecture in the apt sources to allow cross compiling. I'm not fully confident though.

Reply:
Yeah, it's the `sed` stuff I'm querying.  Looks hacky without an explanation.

Reply:
OK, thanks - that's clear now!